File: Crash/Models/BaixaTitulo.cs
using System.ComponentModel.DataAnnotations;
namespace Crash.Models;

public class BaixaTitulo
{
    public int Id { get; set; }
    [Required]
    public required DateTime DataPagamento { get; set; } = DateTime.Now;
    [Required]
    public required decimal ValorBaixado { get; set; }

    // Relacionamentos
    [Required]
    public required int ContaPagarId { get; set; }
    public ContaPagar? ContaPagar { get; set; }
    [Required]
    public required int ContaBancariaId { get; set; }
    public ContaBancaria? ContaBancaria { get; set; }
}

File: FrontendCrash/src/models/BaixaTitulo.ts
import type ContaPagar from "./ContaPagar";
import type ContaBancaria from "./ContaBancaria";

// Interface que representa uma baixa de título (registro de pagamento)
export default interface BaixaTitulo {
    id?: number;
    dataPagamento?: string;
    valorBaixado?: number;
    contaPagarId: number;
    contaBancariaId: number;
    // Objetos de navegação retornados pelo GET
    contaPagar?: ContaPagar;
    contaBancaria?: ContaBancaria;
}
